src/lib/viz/widgets/widget.ts

```diff
@@ -0,0 +1,44 @@
+import { DataView } from '@/data/dataview/dataview';
+import { CartoError } from '@/core/errors/CartoError';
+import { queryDOMElement } from '../utils/dom';
+
+export class Widget {
+  protected element: HTMLElement;
+  protected dataView: DataView;
+
+  constructor(element: string | HTMLElement, dataView: DataView) {
+    const domElement = queryDOMElement(element);
+    validateParameters(domElement, dataView);
+
+    this.element = domElement as HTMLElement;
+    this.dataView = dataView;
+  }
+
+  protected bindEvents() {
+    this.dataView.on('dataUpdate', () => this.updateData());
```

Comment:
so we're assuming here that this is the only and relevant event for any dataview update, right? And this way, we have the 'widget update' logic for all of them....

/cc @manmorjim

Reply:
Yes, that's in fact what I thought. So it doesn't make sense to add a new event. We should rename this one to `dataChange` or merge both events to `dataUpdate` in the DataView.